### PR TITLE
Minor: Add tests for date interval predicate handling

### PR DESCRIPTION
**Pull request by alamb**
~Draft as it builds on https://github.com/apache/arrow-datafusion/pull/4568~

# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/3944
Closes https://github.com/apache/arrow-datafusion/issues/3408

# Rationale for this change

Add teset coverage  https://github.com/apache/arrow-datafusion/issues/3944

# What changes are included in this PR?

New tests

# Are these changes tested?
Yes!
# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/dates.slt [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/dates.slt:1]: This test fails on master like:

```
from test
where d3_date > d2_date + INTERVAL '5 days' AND i_item_desc != 'g';"
Error: SqlLogicTest(query failed: DataFusion error: This feature is not implemented: Unsupported interval argument. Expected string literal, got: BinaryOp { left: Value(SingleQuotedString("5 days")), op: And, right: BinaryOp { left: Identifier(Ident { value: "i_item_desc", quote_style: None }), op: NotEq, right: Value(SingleQuotedString("g")) } }
[SQL] select i_item_desc
from test
where d3_date > d2_date + INTERVAL '5 days' AND i_item_desc != 'g';
at tests/sqllogictests/test_files/dates.slt:62
)
```

**Comment by alamb** [comment]
cc @sarahyurick and @andygrove  (thanks again @sarahyurick  for the fix in sqlparser-rs)

**Comment by alamb** [comment]
cc @sarahyurick and @andygrove  what do you think?

**Comment by sarahyurick** [comment]
Thanks, tests look good to me!

**Pull request merged**